hooks/test_post_gen_project.py

```diff
@@ -49,3 +47,67 @@ def test_copy_journal_template_files_bad(
         match=errormessage,
     ):
         copy_journal_template_files(input, project_dir)
+
+
+# C1: Existing source dir and target dir.
+#  Expect all files in source dir are copied to target dir.
+def test_copy_all_files(user_filesystem, user_repo_files_and_contents):
+    source_dir = user_filesystem / "source-dir"
+    target_dir = user_filesystem / "target-dir"
+    copy_all_files(source_dir, target_dir)
+    for key, value in user_repo_files_and_contents.items():
+        file_path = target_dir / key
+        assert file_path.exists()
+        assert file_path.read_text() == value
+
+
+# C1: An non-existing source dir and an existing target dir.
+#  Expect FileNotFoundError.
+# C2: An empty source dir and an existing target dir.
+#  Expect FlileNotFoundError.
+# C3: Existing source dir and target dir, but there is a file with the same
+#  name found in both dirs. Expect FileExistsError.
+def test_copy_all_files_bad(user_filesystem):
+    # non-existing source directory
+    non_existing_source_dir = user_filesystem / "other-dir"
+    assert not non_existing_source_dir.exists()
+    target_dir = user_filesystem / "target-dir"
+    with pytest.raises(
+        FileNotFoundError,
+        match="Unable to find the source directory: "
+        f"{str(non_existing_source_dir)}. Please leave an issue "
+        "on GitHub.",
+    ):
+        copy_all_files(non_existing_source_dir, target_dir)
+
+    # empty source directory
+    empty_source_dir = user_filesystem / "empty-dir"
+    assert empty_source_dir.exists() and (not any(empty_source_dir.iterdir()))
+    with pytest.raises(
+        FileNotFoundError,
+        match=f"Source directory {str(empty_source_dir)} found "
+        "but it contains no files. Please leave an issue "
+        "on GitHub.",
+    ):
+        copy_all_files(empty_source_dir, target_dir)
+
+    # a file with the same name found in both dirs.
+    gh_source_dir = user_filesystem / "source-dir"
+    target_dir_with_gh_file = user_filesystem / "duplicated-dir"
```

Comment:
ok, looking at this, it is way too complicated.   How about
```
duplicated_file = user_filesystem / target_dir / "usepackage.txt"
assert duplicated_file.is_file()
```
then in the pytest.raises context manager
```
copy_all_files(source_dir, target_dir)
```
The directory names will be changed per the convo about conftest.py.  Also, we can remove `duplicated-dir` from conftest as it won't be needed.

I think this will work, but please check